cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/aws-orchestrator/AwsOrchestrator.service.tsx

```diff
@@ -87,4 +90,71 @@ export class AwsOrchestrator {
             setErrorObject(addErrorInfo(errorObject, e, AWS_FIELDS.NODE_TYPE));
         }
     }
+
+    static initAvailabilityZones(props: AwsOrchestratorProps) {
+        const { awsDispatch, setErrorObject, errorObject } = props;
+        DefaultOrchestrator.initResources<AWSAvailabilityZone>({
+            resourceName: AWS_FIELDS.AVAILABILITY_ZONES,
+            dispatch: awsDispatch,
+            errorObject,
+            setErrorObject,
+            fetcher: () => AwsService.getAwsAvailabilityZones(),
+        });
+    }
+
+    static getAZFieldsForNodeProfile(nodeProfile: string): { [key: string]: AWS_FIELDS }[] {
+        if (nodeProfile === AWS_NODE_PROFILE_NAMES.SINGLE_NODE) {
```

Comment:
Create an array to contain AVAILABILITY_ZONE name and AVAILABILITY_ZONE_NODE_TYPE.
So we can store corresponding availability name and availability node type at the same time.